File: pokeapi-frontend/src/services/berry.service.js
import axios from "axios"

const API_URL = "http://localhost:8080/berry"

class BerryService {
	getSortedBerries(sortField, sortOrder) {
		return axios.get(API_URL, {
			params: {
				sortField: sortField,
				sortOrder: sortOrder,
			},
		})
	}
}

export default new BerryService()

File: pokeapi/src/main/java/com/pokeapi/services/abs/BerryService.java
package com.pokeapi.services.abs;

import java.util.List;

import com.pokeapi.entities.Berry;
import com.pokeapi.enums.SortOrder;

public interface BerryService {

	public abstract List<Berry> getSortedBerries(String sortField, SortOrder sortOrder);

	public abstract void sortBerries(List<Berry> berries, String sortField, SortOrder sortOrder);

}
